lib/src/main/cpp/src/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+project("sgbm_android_core")
+
+set(SOURCE_FILES depth_estimator.cpp)
+set(OPENCV_INSTALL_DIR ${CMAKE_BINARY_DIR}/opencv-install)
+set(OPENCV_LIBRARIES opencv_core opencv_calib3d opencv_imgproc opencv_imgcodecs opencv_flann)
+
+add_library(${PROJECT_NAME} SHARED ${SOURCE_FILES})
+
+target_include_directories(${PROJECT_NAME} PUBLIC ${CMAKE_SOURCE_DIR}/include)
+
+include(ExternalProject)
```

Comment:
Looks like it is better to keep opencv in its separate deps folder with CMakeLists target at building only it, current cmake looks like mess

Look at cmake in the same manner as procedural language -- split functionality into separate cmake files